### PR TITLE
Add `active` event

### PR DESCRIPTION
**Pull request by timgit**
Similar in spirit to a proprosal in #41.  This doesn't necessitate the use case mentioned in that issue, but it would provide a tool which could be used for that purpose.  My motivation for this would be to have a hook into the processing without requiring an external interval to poll a queue instance.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+const events = require('events');
```
Comment (szmarczak) [inline index.js]:
```diff
-const events = require('events');
+const EventEmitter = require('events');
```

File: test.js [hunk 2]
```diff
@@ -402,3 +402,23 @@ test('pause should work when throttled', async t => {
 	delay(2200).then(() => t.deepEqual(result, secondV));
 	await delay(2500);
 });
+
+test('should be an event emitter', t => {
+	const events = require('events');
```
Comment (szmarczak) [inline test.js]: Remove this line and insert at the top:

```js
import EventEmitter from 'events';
```

File: index.js [hunk 3]
```diff
@@ -102,6 +106,7 @@ class PQueue {
 
 	_next() {
 		this._pendingCount--;
+		this.emit('onNext');
```
Comment (szmarczak) [inline index.js]: Need docs for that

Comment (szmarczak) [reply index.js]:
```diff
-this.emit('onNext');
+this.emit('next');
```

**Comment by timgit** [comment]
@szmarczak Updated

**Comment by szmarczak** [comment]
LGTM.

File: test.js [hunk 4]
```diff
@@ -402,3 +403,22 @@ test('pause should work when throttled', async t => {
...
+});
+
+test('should emit next event per item', async t => {
+	const items = [0, 1, 2, 3, 4];
+	let eventCount = 0;
+	const queue = new PQueue();
+
+	queue.on('next', () => eventCount++);
+
+	items.forEach(item => queue.add(() => item));
```
Comment (sindresorhus) [inline test.js]: I would prefer a `for-of` loop here instead.

File: index.js [hunk 5]
```diff
@@ -102,6 +106,7 @@ class PQueue {
 
 	_next() {
 		this._pendingCount--;
+		this.emit('next');
```
Comment (sindresorhus) [inline index.js]: I don't think `next` is a good name for this. It's not descriptive enough when you just read the even subscription code. Is it the `next` event of what? Maybe `active` or `dequeued` or something.

File: readme.md [hunk 6]
```diff
@@ -155,6 +155,11 @@ Number of pending promises.
 
 Whether the queue is currently paused.
 
+## Events
+
```
Comment (sindresorhus) [inline readme.md]: Needs to document that `PQueue` is an EventEmitter subclass and that it has inherited certain methods and link to Node.js docs.

File: readme.md [hunk 7]
```diff
@@ -155,6 +155,11 @@ Number of pending promises.
...
+#### next
+Emitted as each item is processed in the queue for the purpose of tracking progress.
```
Comment (sindresorhus) [inline readme.md]: Needs a code example.

**Comment by timgit** [comment]
@sindresorhus Requested changes pushed.

**Comment by RickWong** [comment]
Does the dependency on the  "events" Node.js package not break browser compatibility? 

Perhaps this could also have been a major version bump.

**Pull request merged**